zegami_sdk/source.py

```diff
@@ -382,7 +382,7 @@ def _upload(self):
         upload_ims['imageset']['source'] = new_source
         # TODO: remove this when backend fixed transfer imageset to use sql storage
         upload_ims['imageset']['imageinfo_storage'] = 'mongodb'
-        payload = json.dumps(upload_ims['imageset'])
+        payload = json.dumps(upload_ims['imageset'], default=lambda o: o.__dict__)
```

Comment:
Not seen this default argument for `dumps` before. What does it do?
Possibly better to have an optional `json` argument for all of the request functions (like we do in cloud) so that the code making these requests doesn't have to worry about this.

Reply:
Is this needed because something in `upload_ims['imageset']` is some sort of class which can't be directly serialised to json? Would be good to know what, as it's not quite obvious from the code above.

Reply:
I spent a few hours but couldn't figure out 😢 , it works when using sdk to create the collection, but throws the error when using the sdk in zegami-ship e2e

Reply:
What was the error text? The only thing I can think of is possibly a different minor version of python might make a difference.

Reply:
File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.9/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type SimpleNamespace is not JSON serializable

Reply:
They are both using python3.7

Reply:
I think the issue is in zegami-ship, `sdk_setup.py` line 58. This converts the dicts that describe the collections into `SimpleNamespace` objects so that the tests can do e.g. `self.coll.v1.name` rather than self.coll['v1']['name'].

I don't think there's any benefit to that beyond a slightly more compact syntax but it makes it a bit awkward to understand what's going on, and also prevents us putting actual dicts into that config. I think it should be refactored to just use a dict.

Reply:
Quickest change is just to put `{'Accept': 'image/jpg'}` in the body of the test function rather than in that config. This PR then shouldn't be needed.

Still doesn't explain why it behaved differently in different contexts 🤷‍♂️

Reply:
Thanks, that works!